include/exadg/grid/grid.h

```diff
@@ -73,6 +74,124 @@ class Grid
     mapping = std::make_shared<dealii::MappingQGeneric<dim>>(data.mapping_degree);
   }
 
+  void
+  create_and_refine_triangulation(
+    GridData const &                                          data,
+    std::vector<int> const &                                  vector_local_refinements,
+    std::function<void(dealii::Triangulation<dim> &)> const & create_coarse_triangulation)
```

Comment:
What is the purpose to have two versions of `create_and_refine_triangulation()`? I would switch the parameters and only have one version..

Reply:
It is described here https://gitlab.lrz.de/exadg/exadg-applications/-/merge_requests/62#note_1620973.

Ideally, refinement would always be done after creating the coarse/serial triangulation. However, this is not possible for certain dealii grid-manipulation tools (working with manifolds). Therefore, I suggest to support both variants in our code.

Reply:
Technically, we could combine both variants to a single function (a boolean parameter decides whether refinement is invoked or not), but I have the fear that it becomes increasingly difficult for the user to understand how to use the code. Note that the main purpose of this PR is to simplify the use of p:f:T for "unexperienced" users.

In principle I agree with you. For now, my suggestion would be that we do not consider the interface of `Grid` fixed once and for all, but get some experience with the new functionality and maybe improve it in the future. Note that there are other functionalities that we want to include such as simplex meshes and reading grids from files. Hence, I expect the `Grid` class to be changed and refactored in the near future.

I have the following idea: We keep two functions in the public interface, but combine the two functions to a single one in the implementation (i.e. we create a private member function).

Reply:
done here https://github.com/exadg/exadg/pull/147/commits/1b8e4c42909bd5d6485c65ad8995f081e7ddfdbe

Reply:
@peterrum Do you agree with the new design (which we can still change in the future)?